src/chart/model/scale.model.ts

```diff
@@ -264,24 +263,41 @@ export class ScaleModel extends ViewportModel {
 		if (initialState.yStart === yStart && initialState.yEnd === yEnd && initialState.zoomY > 0) {
 			return;
 		}
-		super.setYScale(yStart, yEnd, fire);
-		const state = this.export();
-		const constrainedState = this.scalePostProcessor(initialState, state);
 
 		if (this.state.lockPriceToBarRatio) {
-			changeXToKeepRatio(constrainedState, this.zoomXYRatio);
+			this.setLockPriceScale(yStart, yEnd, fire, initialState);
+			return;
+		}
+		this.setYScaleCommon(yStart, yEnd, fire, initialState);
+	}
 
-			this.setXScale(constrainedState.xStart, constrainedState.xEnd);
-			// TODO: rewrite logic for applying constraints to consider both axes, now constraints on Y may not work correctly
+	private setLockPriceScale(yStart: Unit, yEnd: Unit, fire = false, initialState: ViewportModelState) {
```

Comment:
let's call it `setLockedYScale`. we call `price scale` as `y scale` here in `dxcharts-lite`